daemon/logic/session.go

```diff
@@ -52,6 +52,17 @@ func (s *Session) Login(ctx context.Context, creds apitypes.LoginCredential) err
 	return s.engine.session.Set(self.Type, self.Identity, self.Auth, creds.Passphrase(), authToken)
 }
 
+// Refresh retrieves the user's identity and updates the stored copy of the session
```

Comment:
e.g. `PATCH /users/self` or similar on the daemon that also results in update the cache and returns the update `self` to the callee.